### ISSUE TITLE
Add ActNorm.from_data / ActNorm1D.from_data for data-dependent initialisation

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#112]
## Problem / Request

`ActNorm` / `ActNorm1D` expose `loc` and `log_scale` but have no `from_data` factory, even though **data-dependent initialisation is ActNorm's defining feature** (Kingma & Dhariwal 2018): the first batch sets per-channel mean/scale so activations start standardised, which is what makes deep stacks trainable. This is asymmetric with `FixedRotation.from_data`, which already exists for the PCA frame.

With the default `log_scale = 0`, the layer divides every channel by `softplus(0) ≈ 0.69` regardless of the data's scale/shift — so out of the box ActNorm does **not** normalise.

## User Story

> As a user stacking deep flows, I want `ActNorm.from_data(batch)` to mirror `FixedRotation.from_data`, so that data-dependent init is one call instead of a hand-rolled `tree_at` surgery.

## Current workaround (3 lines)

```python
import equinox as eqx
import jax.numpy as jnp
import gauss_flows as gf

def actnorm_from_data(batch):                      # batch: (n, C)
    mu = batch.mean(0)
    sd = batch.std(0)
    inv_softplus = lambda y: jnp.log(jnp.expm1(y)) # softplus^{-1}
    log_scale = inv_softplus(jnp.clip(sd - 1e-5, 1e-6))
    layer = gf.ActNorm(shape=(batch.shape[1],))
    return eqx.tree_at(lambda m: (m.loc, m.log_scale), layer, (mu, log_scale))
```

Verified: output is per-channel mean ≈ 0, std ≈ 1.

## Proposed API

Classmethods mirroring `FixedRotation.from_data`:

```python
ActNorm.from_data(x)     # x: (..., C); loc = per-channel mean, scale = per-channel std
ActNorm1D.from_data(x)   # x: (n, d)
```

handling the softplus inverse and the `+1e-5` floor internally, and reducing over all leading (spatial/batch) axes for the image case.

## Implementation Steps

- [ ] Add `from_data` classmethods in `src/gauss_flows/_src/transforms/bijections/normalization/actnorm.py`
- [ ] Extend `tests/test_from_data_factories.py` (shared with `FixedRotation.from_data`)

## Acceptance Criteria

- [ ] `ActNorm.from_data(batch)` output has per-channel mean ≈ 0, std ≈ 1 (tolerance 1e-5) on the init batch
- [ ] Image case reduces over all leading axes; 1-D case over the batch axis
- [ ] Numerically safe for near-constant channels (the `1e-5` floor)

## Definition of Done

- [ ] Tests pass: `make test`; lint + typecheck green
- [ ] Docstrings (Google-style, **Shape:** + **Example:**)

## Context

Surfaced writing the Gaussianization tutorial (Part 2, notebook `02_rotations/03_conv1x1_actnorm`).

## Relationships

- Parent (theme epic): #125
- Blocked by: —
- Related: `FixedRotation.from_data` (pattern to mirror); #72 (Haar-sampled rotation init — same "principled initialisation" family)
